mmcv/runner/hooks/logger/text.py

```diff
@@ -142,10 +142,22 @@ def _round_float(self, items):
 
     def log(self, runner):
         log_dict = OrderedDict()
-        # training mode if the output contains the key "time"
-        mode = 'train' if 'time' in runner.log_buffer.output else 'val'
-        log_dict['mode'] = mode
-        log_dict['epoch'] = runner.epoch + 1
+
+        if 'time' in runner.log_buffer.output and runner.mode == 'train':
+            # normal train mode
+            log_dict['mode'] = 'train'
+            log_dict['epoch'] = runner.epoch + 1
+        elif runner.mode == 'val':
+            # normal val mode
+            # runner.epoch += 1 has been done before val workflow
+            log_dict['mode'] = 'val'
+            log_dict['epoch'] = runner.epoch
+        else:
+            # Evaluation hook during train mode
+            # runner.epoch += 1 should be done since training has ended
+            log_dict['mode'] = 'val'
+            log_dict['epoch'] = runner.epoch + 1
+
```

Comment:
It may be better to simplify the logic as follows:
```
        if runner.mode == 'train':
            log_dict['mode'] = 'train' if 'time' in runner.log_buffer.output \
                else 'val'
            log_dict['epoch'] = runner.epoch + 1
        elif runner.mode == 'val':
            # normal val mode
            # runner.epoch += 1 has been done before val workflow
            log_dict['mode'] = 'val'
            log_dict['epoch'] = runner.epoch
        else:
            raise ValueError

```